lib/views/message-count.js

```diff
@@ -0,0 +1,44 @@
+/** @babel */
+/** @jsx etch.dom */
+
+import etch from 'etch'
+import { CompositeDisposable } from 'atom'
+import MessageIcon from './message-icon'
+
+export default class MessageCount {
+  disposables = new CompositeDisposable()
+
+  constructor (properties = { type: 'error' }) {
+    this.properties = properties
+    etch.initialize(this)
+    this.disposables.add(latex.log.onMessages(() => this.update()))
+  }
+
+  async destroy () {
+    await etch.destroy(this)
+    this.disposables.dispose()
+  }
+
+  render () {
+    const loggingLevel = atom.config.get('latex.loggingLevel')
+
+    if ((loggingLevel === 'error' && this.properties.type !== 'error') ||
+      (loggingLevel === 'warning' && this.properties.type === 'info')) {
```

Comment:
Maybe we can extract two booleans here with descriptive names to make it a little bit clearer what we're checking?